Tasks/DeployAzureResourceGroup/SPNCreation.ps1

```diff
@@ -3,6 +3,9 @@ param
     [Parameter(Mandatory=$true, HelpMessage="Enter Azure Subscription name. You need to be Subscription Admin to execute the script")]
     [string] $subscriptionName,
 
+	[Parameter(Mandatory=$true, HelpMessage="Provide a name for the SPN application that you would create")]
```

Comment:
Can we make this an optional parameter? For scenarios you mentioned, users can supply app name. But for most cases where it works, users can worry one less argument.